src/pydisagg/ihme/splitter/sex_splitter.py

```diff
@@ -152,10 +153,14 @@ def parse_population(
         data_with_population = self._merge_with_population(
             data, male_population, "m_pop"
         )
+
         data_with_population = self._merge_with_population(
             data_with_population, female_population, "f_pop"
         )
 
+        validate_population_data(
+            data_with_population, ["m_pop", "f_pop"], self.data.index, name
+        )
```

Comment:
I believe we want to add `validate_noindexdiff` as we do in age splitter in case some of the population are missing.